### ISSUE TITLE
Private role variables raise var-naming[no-role-prefix] warning

### ISSUE THREAD

**Issue by netsandbox** [ansible/ansible-lint#3760]
##### Summary

Private role variables (begin with an underscore) raise var-naming[no-role-prefix] warning.

Ansible is not using the concept of private variables: https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html

> Variable names can begin with an underscore. In many programming languages, variables that begin with an underscore are private. This is not true in Ansible. Variables that begin with an underscore are treated exactly the same as any other variable. Do not rely on this convention for privacy or security.

But I have often seen usage of private variables in roles, e.g. https://github.com/geerlingguy/ansible-role-redis/blob/master/vars/RedHat.yml#L2C3-L2C3
(by the way, I don't know why @geerlingguy uses here two underscores instead of one.)

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.20.0 using ansible-core:2.14.10 ansible-compat:4.1.10 ruamel-yaml:0.17.32 ruamel-yaml-clib:0.2.7
```

##### STEPS TO REPRODUCE

Create a role `my_role` with a `_my_role_var` variable.

##### Desired Behavior

Role Variables which have the role name as prefix and begin with an underscore (multiple underscores?) should not raise a `var-naming[no-role-prefix]` warning.

##### Actual Behavior

```
var-naming[no-role-prefix]: Variables names from within roles should use my_role_ as a prefix. (vars: _my_role_var)
```

**Comment by geerlingguy** [comment]
I just use `_` or `__` as a convention (from way back in PHP days), meaning "this variable is for internal use... if you try overwriting it, and there's an issue, not my problem".

I would never try to indicate any additional security, it's just an informal way of saying "this is not the variable you're looking for"

**Comment by ssbarnea** [comment]
@geerlingguy @netsandbox Too prevent accidental shadowing of variables, especially when nesting roles, what do you think about ignoring the underscores but still requiring the prefix.

Example: `_result` would not be allowed, but `_myrole_result` would be, assuming that prefix is `myrole`.

At least our teams thinks that this should be a good compromise.

**Comment by geerlingguy** [comment]
@ssbarnea yeah, that would make more sense.

**Comment by netsandbox** [comment]
@ssbarnea that is actually what I meant with:
> Desired Behavior
Role Variables which have the role name as prefix and begin with an underscore (multiple underscores?) should not raise a var-naming[no-role-prefix] warning.

**Comment by dnapier** [comment]
So for example, a role for helm would have variables like `helm_` or `_helm`? Seems like either should be fine, but the former is raising the lint issue for me.

> @ssbarnea that is actually what I meant with:
> 
> > Desired Behavior
> > Role Variables which have the role name as prefix and begin with an underscore (multiple underscores?) should not raise a var-naming[no-role-prefix] warning.
